src/Server/Services/Disk/StoageInfoProvider.cs

```diff
@@ -0,0 +1,89 @@
+/*
+ * Apache License, Version 2.0
+ * Copyright 2019-2021 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Nvidia.Clara.DicomAdapter.Configuration;
+using System;
+using System.IO.Abstractions;
+
+namespace Nvidia.Clara.DicomAdapter.Server.Services.Disk
+{
+    public interface IStorageInfoProvider
+    {
+        bool HasSpaceAvailableToStore { get; }
+        bool HasSpaceAvailableForExport { get; }
+        bool HasSpaceAvailableToRetrieve { get; }
+        long AvailableFreeSpace { get; }
+    }
+
+    public class StorageInfoProvider : IStorageInfoProvider
+    {
+        private const long OneGB = 1000000000;
```

Comment:
a gigabyte is 1024 * 1024 * 1024, or 1,073,741,824 bytes; not 1000000000 bytes.

Reply:
`GB` (base 10) is 1000^3 and `GiB` (base 2) is 1024^3.